src/btree/bt_ret.c

```diff
@@ -216,7 +218,9 @@ __wt_value_return_upd(WT_SESSION_IMPL *session,
 	 * If we hit the end of the chain, roll forward from the update item we
 	 * found, otherwise, from the original page's value.
 	 */
-	if (upd == NULL && !skipped_birthmark) {
+	if (skipped_birthmark)
+		WT_ERR(__wt_buf_set(session, &cursor->value, "", 0));
```

Comment:
This case is already worrying for a tombstone (we should return `WT_NOTFOUND` so if we get here it's probably masking an error).

I understand the bug fix, but would it be reasonable to assert that neither case ever happens?

Reply:
Also, as mentioned above, if we find a visible update, we should take it, and only check whether we have skipped a birthmark if `upd == NULL`.

Reply:
> This case is already worrying for a tombstone (we should return WT_NOTFOUND so if we get here it's probably masking an error).
> I understand the bug fix, but would it be reasonable to assert that neither case ever happens?

I don't understand what you're trying to say, here.